gadget/layout.go

```diff
@@ -284,6 +309,10 @@ func LayoutVolume(volume *Volume, opts *LayoutOptions) (*LaidOutVolume, error) {
 			farthestEnd = end
 		}
 
+		// Set appropriately label and type details
+		// TODO: set this in layoutVolumeStructures in the future.
+		setOnDiskLabelAndTypeInLaidOuts(opts.EncType, structures)
```

Comment:
this code doesn't seem exercised by the unit test, as I don't see any new test passing an EncType different from ...None

Reply:
Tests added now